virtcontainers/device/drivers/vfio.go

```diff
@@ -64,14 +65,14 @@ func (device *VFIODevice) Attach(devReceiver api.DeviceReceiver) error {
 	}
 
 	// Pass all devices in iommu group
-	for _, deviceFile := range deviceFiles {
+	for i, deviceFile := range deviceFiles {
 		//Get bdf of device eg 0000:00:1c.0
 		deviceBDF, err := getBDF(deviceFile.Name())
 		if err != nil {
 			return err
 		}
 		vfio := &config.VFIODev{
-			ID:  utils.MakeNameID("vfio", device.DeviceInfo.ID, maxDevIDSize),
+			ID:  utils.MakeNameID("vfio", device.DeviceInfo.ID+strconv.Itoa(i), maxDevIDSize),
```

Comment:
Good find! This is a true bug, thanks!

Can you help to modify `device.DeviceInfo.ID+strconv.Itoa(i)` to `device.ID+strconv.Itoa(i)` ? I just realized `DeviceInfo` isn't necessary.

Reply:
@WeiZhang555 - oh! We have two fields with the same information? Do we need to remove `GenericDevice.ID` so that we only have `GenericDevice.DeviceInfo.ID` then?

If we need to keep both:

- We should add a comment explaining why.
- We should have tests to ensure both `ID` values are the same.

Reply:
`DeviceInfo` is input data for creating a new device, `GenericDevice.ID` is the real device ID made by each device driver, in my original refactor, the `DeviceInfo` is designed to be throwed away and not stored, and `DeviceInfo` should be only for one-time use. 
But currently so many places are using `DeviceInfo`, and it is involved deeply, that's why I didn't do it.

But you remind me that it's better to do it earlier :smile: I'll take a look at it.